### DISCUSSION TITLE
Request / Response with CorrelationData

### DISCUSSION THREAD

**Discussion by guy-avery** [dotnet/MQTTnet#1244] [Q&A] [answered]
Hi - being fairly new to the MQTT protocol I was doing some reading on V5 of the protocol to understand how the request / response mechanism works. I have implemented the response / request pattern with this MQTTNet lib along with the Rpc extension package. However it does not appear to use the correlation data and therefore does not appear to truly conform to the V5 specification. 
Firstly have I got that right?
If so, is this just for backward compatibility so RPC calls can be made on earlier versions of the protocol? 
Are there any plans to enhance the Rpc extension to use the V5 pattern?

**Comment by chkr1011** [comment, marked as answer]
Hi, the RPC extension adds RCP support for all protocol versions. Including a async wrapper so that you can write a simple await instead of having a handler somewhere. 

There are currently no active plans for the RPC extension but you are welcome to create a pull request with the required changes.
